src/main/java/com/team766/framework3/ContextImpl.java

```diff
@@ -405,96 +345,119 @@ public void yield() {
     }
 
     @Override
-    public LaunchedContext startAsync(final RunnableWithContext func) {
-        return new ContextImpl(func, this);
+    public LaunchedContext startAsync(final Procedure procedure) {
```

Comment:
I am also checking for reservation conflicts with the current Context, so there is some cohesion here.

Open to suggestions

Reply:
I found no use cases for `startAsync` in the 2024/Gatorade code. I left it in because WPILib also offers a similar capability, though they also highly recommend only using it as a last resort https://docs.wpilib.org/en/stable/docs/software/commandbased/command-compositions.html#scheduling-other-commands

I'm find with moving this out of `Context` to make it inconvenient / help break habits of people used to using MF2